Docs: Getting started chapter on MicroPython internals

nanjekyejoannah opened the pull request:
MicroPython internals docs.

Getting started chapter.

cc @mattytrentini @jimmo

nanjekyejoannah commented:
Closing and opening to re-trigger CI.

stinos commented on docs/develop/gettingstarted.rst:
```diff
@@ -0,0 +1,285 @@
+
+.. code-block:: console
+
+   $ gcc --version
```
It could be confusing that this will only work after the installation listed below, more so for building of the cross-compiler. Also Python (best to recommend Python 3 I guess) needs to be installed.

nanjekyejoannah replied:
What are you suggesting then? That we move details of this dependency to before building the cross-compiler?

stinos replied:
Definitely before it, yes, since those are all prerequisite. As far as I know people who read this just want to get things working so it makes sense that it's a document which you start from the top end read to the bottom, following steps in the order presented.

Actually there could even a separate section which bundles the prerequisites, i.e. the ones mentioned now, and Python, and git as well because it's not really different from the other applications needed.

dlech commented:
> Closing and opening to re-trigger CI.

This failure was fixed in https://github.com/micropython/micropython/pull/6451/commits/5c503de52166f6ebd4ef59b58b130a730782823d. You can rebase your branch on the current master branch to pull in this change.

nanjekyejoannah commented:
huh fixing this rebase mishap :(

nanjekyejoannah commented:
I think the branch `docs-dev` needs to be updated to match the master branch. The rebase is working as expected.

mcauser commented on docs/develop/gettingstarted.rst:
```diff
@@ -0,0 +1,285 @@
+After the above configuration, your setup should be similar to this:
+
+.. code-block:: console
+
```
Maybe add a `$ git remote -v` here.

Commit:
docs/develop: getting started chapter on micropython internals

nanjekyejoannah commented:
I opened a new PR that supersedes this here : https://github.com/micropython/micropython/pull/6543

I addressed the comments please check @stinos if I addressed the dependencies part well.
